### ISSUE TITLE
[onert] Remove #ifdef ONERT_TRAIN

### ISSUE THREAD

**Issue by glistening** [Samsung/ONE#12293]
We need to remove `#ifdef ONERT_TRAIN` as much as possible.
Hopefully, remove the whole occurrences.

They are introduced when we're doing prototyping and in a hurry.

I am not the author of many of #ifdef ONERT_TRAIN.
It needs the help of the original authors.

- [ ] getBufSize (nnfw_api_internal.cc)
- [ ] ...

**Comment by glistening** [comment]
I would like to start with seemingly simple one.

@jyoungyun May I ask why `getBufSize(...)` needs to be enclosed with `#ifdef ONERT_TRAIN`.

https://github.com/Samsung/ONE/blob/11620233e1a303dcc60f48fc7103cb55d84270cc/runtime/onert/api/src/nnfw_api_internal.cc#L199-L221

It is not training-dependent code.

We may move the following existing code into somewhere.

https://github.com/Samsung/ONE/blob/11620233e1a303dcc60f48fc7103cb55d84270cc/tests/tools/onert_run/src/nnfw_util.cc#L23-L47

**Comment by jyoungyun** [comment]
> May I ask why `getBufSize(...)` needs to be enclosed with `#ifdef ONERT_TRAIN`.

When we first started ONERT training, we decided to use `ONERT_TRAIN` definition if the code is used for only ONERT train. So I wrote the code like this.

At the beginning of the project, there were many concerns about the direct merge of ONERT training code   into ONERT from the perspective of file size, memory size, release criteria and performance. Is there anything more concerns about these?
